### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by jitsedesmet**
This pull request doesn't provide any new features. It adds some new tests to increase coverage. And asks some questions on certain tests.

### CODE REVIEW

File: test/integration/functions/onDates.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const d2 = date('2010-12-21T15:38:02-10:00');
+  const d3 = date('2010-12-21T15:38:02-11:00');
+  const d4 = date('2010-12-21T15:38:02+11:30');
+
+  describe('evaluation of \'timezone\' like', () => {
+    runTestTable({
+      operation: 'TIMEZONE',
+      notation: Notation.Function,
+      arity: 1,
+      // TODO: PR: is this first option really what we want? I'm pretty sure we want PT0S
```
Comment (wschella) [inline test/integration/functions/onDates.test.ts:16]: I have no clue anymore, let's find a reference and change as you want.

File: test/integration/functions/onStrings.test.ts [hunk 2]
```diff
@@ -35,6 +35,18 @@ describe('string functions', () => {
     });
   });
 
+  // TODO: PR: I put this here for coverage, I don't know if it's correct
```
[the remark this answers is not in the document]
Comment (wschella) [reply test/integration/functions/onStrings.test.ts:38]: If the flag is defined as such in the spec, this seems correct?

File: test/integration/functions/op.bound.test.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    operation: 'BOUND',
+    config: { type: 'sync', bindings, config: {}},
+    arity: 1,
+    aliases: bool,
+    notation: Notation.Function,
+    testTable: `
+    '?x' = true
+    '?y' = false
+    `,
+    // TODO: PR: this is handled in a module, do we need the check on line 37 of SpecialFunctions?
```
[the remark this answers is not in the document]
Comment (wschella) [reply test/integration/functions/op.bound.test.ts:23]: I don't know if what you mean with "this is handled in a module". But we do need the check on line 37, as we need to make sure were working with a variable, e.g. literals should throw errors.

File: test/integration/functions/op.sameTerm.test.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    operation: 'sameTerm',
+    config: { type: 'sync', bindings, config: {}},
+    arity: 2,
+    aliases: bool,
+    notation: Notation.Function,
+    testTable: `
+    ?a ?a = true
+    ?a ?b = true
+    ?a ?c = false
+    ?a ?d : false
```
Comment (wschella) [inline test/integration/functions/op.sameTerm.test.ts:27]: Is this supposed to be a `:`?

File: test/integration/functions/op.unary.test.ts [hunk 5]
```diff
@@ -30,14 +30,20 @@ describe('unary functions', () => {
   });
 
   describe('evaluation of \'+ (unary)\' like', () => {
+    // TODO: PR why does this test not run over the unary plus operator?
+    //  Also add '"-6"^^<${TypeURL.XSD_NEGATIVE_INTEGER}>' = -6i when it does
```
[the remark this answers is not in the document]
Comment (wschella) [reply test/integration/functions/op.unary.test.ts:34]: I don't understand the question @jitsedesmet is the unary plus operator not applied?

**Review by wschella — COMMENTED** [review]
Looks good, very useful extra tests. Conflicts should ofc be resolved, something weird happened with the changed files I think, it's a lot more than just tests.

File: test/integration/functions/op.sameTerm.test.ts (hunk 4 above)
Comment (jitsedesmet) [reply test/integration/functions/op.sameTerm.test.ts:27]: > Is this supposed to be a :?

Yes, thank you.

> I don't know if what you mean with "this is handled in a module". But we do need the check on line 37, as we need to make sure were working with a variable, e.g. literals should throw errors.

I think what I meant with this is that the parser checks this. I'm not sure tho.

**Comment by jitsedesmet** [comment]
> I don't understand the question @jitsedesmet is the unary plus operator not applied?

Indeed, the operator is not applied. Is there any reason for this that you ca remember?

**Comment by wschella** [comment]
> > I don't understand the question @jitsedesmet is the unary plus operator not applied?
> 
> Indeed, the operator is not applied. Is there any reason for this that you ca remember?

Not really.

**Comment by Tpt** [comment]
@jitsedesmet Hi! I am working to improve sparqlee code coverage. Would you be ok if I try to move forward your pull request?

**Comment by jitsedesmet** [comment]
Not at all. Go ahead. I would be happy if it can be of any use!

**Comment by Tpt** [comment]
Great! Thank you!

**Pull request closed without merging**